apps/crn-server/test/routes/research-outputs.route.test.ts

```diff
@@ -120,6 +138,124 @@ describe('/research-outputs/ route', () => {
         researchOutputId,
       );
     });
+    test('Should return 404 when you are trying to get a draft and are not part of the team', async () => {
+      userMockFactory.mockReturnValueOnce(createUserResponse());
+
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        published: false,
+        teams: [{ id: 'nop', displayName: 'team 1' }],
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(
+        researchOutputResponse,
+      );
+
+      const response = await supertest(app).get('/research-outputs/123');
+
+      expect(response.status).toEqual(404);
+    });
+    test('Should return 404 when you are trying to get a draft and are not part of the working group', async () => {
+      userMockFactory.mockReturnValueOnce(createUserResponse());
+
+      const researchOutputResponse: ResearchOutputWorkingGroupResponse = {
+        ...getResearchOutputResponse(),
+        published: false,
+        teams: [{ id: 'nop', displayName: 'team 1' }],
+        workingGroups: [{ id: 'nop', title: 'wg 1' }],
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(
+        researchOutputResponse,
+      );
+
+      const response = await supertest(app).get('/research-outputs/123');
+
+      expect(response.status).toEqual(404);
+    });
+    test('Should return 200 when you are ASAP staff member and are trying to get a draft', async () => {
+      userMockFactory.mockReturnValueOnce({
+        ...createUserResponse(),
+        role: 'Staff',
```

Comment:
same, add the teams, for what i know this test could be passing not because of the role but because of some team that the user belongs to